### PR TITLE
DAT-411 - Add InvoiceDate and PaymentDate properties

### PR DESCRIPTION
**Pull request by jhoffman-ms**
Add two new properties ```InvoiceDate``` and ```PaymentDate``` in the CreateInvoice and UpdateInvoice endpoints.

These properties are used to set the dates in the invoices and payments when it send to SAP.

**Task:** [DAT-411](https://makingsense.atlassian.net/jira/software/projects/DAT/boards/62?selectedIssue=DAT-411)

### CODE REVIEW

**Review by bamaro-dev — APPROVED** [review]
Super great!

**Comment by nreal** [comment]
👍

File: Doppler.Sap/Mappers/Billing/BillingForArMapper.cs [hunk 1]
```diff
@@ -36,18 +36,21 @@ public bool CanMapSapSystem(string sapSystem)
 
         public SapSaleOrderModel MapDopplerBillingRequestToSapSaleOrder(BillingRequest billingRequest)
         {
+            var invoiceDate = billingRequest.InvoiceDate ?? _dateTimeProvider.GetDateByTimezoneId(_dateTimeProvider.UtcNow, _timezoneConfig.InvoicesTimeZone);
```
Comment (nreal) [inline Doppler.Sap/Mappers/Billing/BillingForArMapper.cs:39]: what do you think add in this line the command `.ToString("yyyy-MM-dd")` ?

File: Doppler.Sap/Mappers/Billing/BillingForUsMapper.cs [hunk 2]
```diff
@@ -65,6 +65,9 @@ public bool CanMapSapSystem(string sapSystem)
 
         public SapSaleOrderModel MapDopplerBillingRequestToSapSaleOrder(BillingRequest billingRequest)
         {
+            var date = billingRequest.InvoiceDate ?? _dateTimeProvider.GetDateByTimezoneId(_dateTimeProvider.UtcNow, _timezoneConfig.InvoicesTimeZone);
```
Comment (nreal) [inline Doppler.Sap/Mappers/Billing/BillingForUsMapper.cs:68]: what do you think add in this line the command .ToString("yyyy-MM-dd") ?

File: Doppler.Sap/Mappers/Billing/BillingForUsMapper.cs [hunk 3]
```diff
@@ -156,13 +160,15 @@ public SapSaleOrderModel MapDopplerBillingRequestToSapSaleOrder(BillingRequest b
             return sapSaleOrder;
         }
 
-        public SapIncomingPaymentModel MapSapIncomingPayment(int docEntry, string cardCode, decimal docTotal, string transferReference)
+        public SapIncomingPaymentModel MapSapIncomingPayment(int docEntry, string cardCode, decimal docTotal, string transferReference, DateTime? paymentDate)
         {
+            var date = paymentDate ?? _dateTimeProvider.GetDateByTimezoneId(_dateTimeProvider.UtcNow, _timezoneConfig.InvoicesTimeZone);
```
Comment (nreal) [inline Doppler.Sap/Mappers/Billing/BillingForUsMapper.cs:165]: what do you think add in this line the command .ToString("yyyy-MM-dd") ?

File: Doppler.Sap/Mappers/Billing/BillingForArMapper.cs (hunk 1 above)
Comment (jhoffman-ms) [reply Doppler.Sap/Mappers/Billing/BillingForArMapper.cs]: Done

File: Doppler.Sap/Mappers/Billing/BillingForUsMapper.cs (hunk 2 above)
Comment (jhoffman-ms) [reply Doppler.Sap/Mappers/Billing/BillingForUsMapper.cs]: Done

File: Doppler.Sap/Mappers/Billing/BillingForUsMapper.cs (hunk 3 above)
Comment (jhoffman-ms) [reply Doppler.Sap/Mappers/Billing/BillingForUsMapper.cs]: Done

**Pull request merged**